app/javascript/single-file-upload.js

```diff
@@ -0,0 +1,101 @@
+window.addEventListener("DOMContentLoaded", (event) => {
+  const fileInput = document.querySelector('.govuk-file-upload');
+  const maxFileSize = fileInput.dataset.maxSize;
+  const errorHeading = fileInput.dataset.errorHeading;
+  const errorMessage = fileInput.dataset.sizeErrorMessage;
+  const feedback = $(".single-file-upload__message");
+  const saveButtons = document.querySelectorAll('button[type="submit"]');
+
+  if(fileInput){
+    /*
+      logic for file progress indicator
+      and uploaded files list
+    */
+    fileInput.addEventListener('change', function(event) {
+        if(fileInput.files.length > 0){
+          handleUploadList(fileInput);
+        }
+    });
+
+    /*
+      handle checking file size, if exceeds limit,
+      refresh with error shown and move page to file upload section.
+    */
+    if (saveButtons) {
```

Comment:
Same as above here

Reply:
see comment above

Reply:
updated this for loop suggestion but see [this comment ](https://github.com/ministryofjustice/laa-submit-crime-forms/pull/1296#discussion_r1839912529)re: the other for loop suggestion